packages/web-forms/src/lib/cache/image.ts

```diff
@@ -0,0 +1,19 @@
+import type { JRResourceURL } from '@getodk/common/jr-resources/JRResourceURL.ts';
+import { type ObjectURL, revokeObjectURL } from '@getodk/common/lib/web-compat/url.ts';
+
+type ImageURL = string;
+
+const image = new Map<ImageURL, ObjectURL>();
+
+export const getCachedImage = (url: JRResourceURL): ObjectURL | null => {
+	return image.get(url.href) ?? null;
+};
+
+export const cacheImage = (url: JRResourceURL, blob: ObjectURL): void => {
+	image.set(url.href, blob);
+};
+
+export const clearImageCache = (): void => {
```

Comment:
I'll make a service for clearing all caches/memory when the form is unmounted/destroyed, or something around that idea